backend/cmm_helpers.ml

```diff
@@ -826,9 +826,7 @@ let get_header ptr dbg =
      data race on headers. This saves performance with ThreadSanitizer
      instrumentation by avoiding to instrument header loads. *)
   Cop
-    ( (* BACKPORT BEGIN mk_load_immut Word_int, *)
-      mk_load_mut Word_int,
-      (* BACKPORT END *)
+    ( mk_load_immut Word_int,
```

Comment:
I believe this is safe, but we should probably completely remove the `obj_set_tag` and `obj_truncate` functions (currently they are not declared in `obj.ml`, but remain available for anyone to declare as external).

Reply:
I'd like to think about this a bit more before we merge (@xclerc let's discuss)